packages/core/src/new-api/internal/deployment-loader/memory-deployment-loader.ts

```diff
@@ -0,0 +1,55 @@
+import fs from "fs";
+
+import { Artifact, ArtifactResolver, BuildInfo } from "../../types/artifact";
+import { DeploymentLoader } from "../../types/deployment-loader";
+import { Journal } from "../../types/journal";
+import { MemoryJournal } from "../journal/memory-journal";
+
+export class MemoryDeploymentLoader implements DeploymentLoader {
+  public journal: Journal;
+
+  private _deployedAddresses: { [key: string]: string };
+
+  constructor(private _artifactResolver: ArtifactResolver) {
+    this.journal = new MemoryJournal();
+    this._deployedAddresses = {};
+  }
+
+  public async loadArtifact(storedArtifactPath: string): Promise<Artifact> {
+    const json = await fs.promises.readFile(storedArtifactPath);
+
+    const artifact = JSON.parse(json.toString());
+
+    return artifact;
+  }
+
+  public async initialize(): Promise<void> {}
+
+  public async recordDeployedAddress(
+    futureId: string,
+    contractAddress: string
+  ): Promise<void> {
+    this._deployedAddresses[futureId] = contractAddress;
+  }
+
+  public async storeArtifact(
+    _futureId: string,
+    artifact: Artifact
+  ): Promise<string> {
+    const artifactPath = await this._artifactResolver.resolvePath(
```

Comment:
This is a quirk of wanting to return the file path of the original contract artifact under `./artifacts` but starting with the `artifact` object.

An alternative that removes this dep on the artifactResolver would be to pass in both artifact and its original path.

Reply:
The resolvePath function on artifactResolver exists solely for this use case.